### PR TITLE
Support passing an iam role AND a k8s service account

### PR DESCRIPTION
**Pull request by katrogan**
Signed-off-by: Katrina Rogan <katroganGH@gmail.com>

# TL;DR
Support passing an iam role AND a k8s service account

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
User suggested feature

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1057

## Follow-up issue
_NA_

### CODE REVIEW

File: protos/flyteidl/admin/launch_plan.proto [hunk 1]
```diff
@@ -60,10 +60,8 @@ message LaunchPlanList {
 // Defines permissions associated with executions created by this launch plan spec.
 // Deprecated.
 message Auth {
-    oneof method {
-        string assumable_iam_role = 1;
-        string kubernetes_service_account = 2;
-    }
+    string assumable_iam_role = 1;
```
Comment (EngHabu) [inline protos/flyteidl/admin/launch_plan.proto:63]: while we are at it... docs?
also maybe elaborate on where each will be used?

Comment (kumare3) [reply protos/flyteidl/admin/launch_plan.proto:63]: Yes please

Comment (katrogan) [reply protos/flyteidl/admin/launch_plan.proto:65]: done, although technically this message is deprecated 🙃

**Review by kumare3 — APPROVED** [review]

File: protos/flyteidl/admin/launch_plan.proto (hunk 1 above)
Comment (EngHabu) [reply protos/flyteidl/admin/launch_plan.proto:65]: OMG I just saw that... yes yes sorry had you do this...

**Review by EngHabu — APPROVED** [review]

**Comment by bstadlbauer** [comment]
Hi @katrogan 

We've just upgraded to `flyteidl` from version 0.18.46 to 0.18.50 and noticed that after registering a launchplan and querying it afterwards using `flyte-cli list-launch-plan-versions` does give us the following error:

```
Traceback (most recent call last):
  File "/srv/workflows/mlops/.venv/bin/flyte-cli", line 8, in <module>
    sys.exit(_flyte_cli())
  File "/srv/workflows/mlops/.venv/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/srv/workflows/mlops/.venv/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/srv/workflows/mlops/.venv/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/srv/workflows/mlops/.venv/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/srv/workflows/mlops/.venv/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/srv/workflows/mlops/.venv/lib/python3.7/site-packages/flytekit/clis/flyte_cli/main.py", line 1014, in list_launch_plan_versions
    sort_by=_admin_common.Sort.from_python_std(sort_by) if sort_by else None,
  File "/srv/workflows/mlops/.venv/lib/python3.7/site-packages/flytekit/clients/friendly.py", line 431, in list_launch_plans_paginated
    [_launch_plan.LaunchPlan.from_flyte_idl(pb) for pb in lp_list.launch_plans],
  File "/srv/workflows/mlops/.venv/lib/python3.7/site-packages/flytekit/clients/friendly.py", line 431, in <listcomp>
    [_launch_plan.LaunchPlan.from_flyte_idl(pb) for pb in lp_list.launch_plans],
  File "/srv/workflows/mlops/.venv/lib/python3.7/site-packages/flytekit/models/launch_plan.py", line 386, in from_flyte_idl
    spec=LaunchPlanSpec.from_flyte_idl(pb2_object.spec),
  File "/srv/workflows/mlops/.venv/lib/python3.7/site-packages/flytekit/models/launch_plan.py", line 237, in from_flyte_idl
    auth_role = _common.AuthRole.from_flyte_idl(pb2.auth_role)
  File "/srv/workflows/mlops/.venv/lib/python3.7/site-packages/flytekit/models/common.py", line 458, in from_flyte_idl
    assumable_iam_role=pb2_object.assumable_iam_role if pb2_object.HasField("assumable_iam_role") else None,
ValueError: Can't test non-optional, non-submessage field "AuthRole.assumable_iam_role" for presence in proto3.
```

Could this be related to this change?

**Comment by katrogan** [comment]
hi @bstadlbauer the latest flytekit release v0.18.1 should have the fix. thanks again for your patience!

**Pull request merged**